Use Cache

darekrossman opened the pull request:
This PR introduces the new `use cache` directive, replace `unstable_cache`, as well as some minor rearranging of files for better organization.

darekrossman committed:
refactor: remove unused instrumentation file and update imports

- Deleted the unused instrumentation.ts file.
- Updated imports in various components to use the new sfcc structure.
- Added environment variable validation in sitemap and adjusted related components.

darekrossman committed:
refactor: update imports and improve consistency across components

- Refactored import paths to align with the new sfcc structure in multiple components.
- Standardized string formatting in the search page for improved readability.
- Adjusted schemas and utility imports for better organization.

darekrossman committed:
fix: enable ppr in next.config.ts

darekrossman committed:
chore: update Next.js version and disable experimental PPR feature

- Updated Next.js version from 15.2.0-canary.67 to 15.3.0-canary.13 in package.json and pnpm-lock.yaml.
- Disabled the experimental PPR feature in next.config.ts for improved stability.

ah100101 commented on app/search/[collection]/page.tsx:
```diff
@@ -3,8 +3,8 @@ import { notFound } from "next/navigation";
 
 import Grid from "components/grid";
 import ProductGridItems from "components/layout/product-grid-items";
+import { defaultSort, sorting } from "lib/constants";
```
I think it would be best for for this to actually be in lib/sfcc/constants

ah100101 commented on lib/constants.ts:
```diff
@@ -16,6 +10,10 @@ export type SortFilterItem = {
   reverse: boolean;
 };
 
+export const storeCatalog = {
+  ids: "mens,womens,newarrivals,top-seller",
+};
+
```
Let's move storeCatalog to sfcc/constants

ah100101 commented on lib/sfcc/constants.ts:
```diff
@@ -1,50 +1,34 @@
 
-export const storeCatalog = {
-  ids: 'mens,womens,newarrivals,top-seller'
-};
```
I actually think these were OK in sfcc since it requires these being created in sfcc

darekrossman committed:
refactor: migrate constants to sfcc structure and update imports

- Moved sorting and defaultSort constants to a new sfcc/constants.ts file for better organization.
- Updated import paths across multiple components to reflect the new structure.
- Adjusted currency code handling in the carousel component for accuracy.
